Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/openSAML/extraction/AttributeHelperTest.java

```diff
@@ -0,0 +1,156 @@
+     */
+    @Test
+    public final void testExtractAttributeValueString() {
+
+        // Test for exceptions if attribute is Null
+        AttributeHelper helper = new AttributeHelper();
+        Attribute attrib = mock(Attribute.class);
+        helper.extractAttributeValueString(attrib);
+
+    }
+
+    @Test
+    public final void testExtractNhinCodedElement() {
+
+        // Test for exception when Attrib value passed in Attribute is Null
+        AttributeHelper helper = new AttributeHelper();
+        Attribute attrib = mock(Attribute.class);
+        List<XMLObject> attrVals = new ArrayList<>();
+        when(attrib.getAttributeValues()).thenReturn(attrVals);
+        String CodeId = null;
+        helper.extractNhinCodedElement(attrib, CodeId);
+
+        // Test for exception when Attrib value is not null and child node of Attrib value is set
+        final String purposeCode2 = "purposeCode";
+        final String purposeSystem2 = "purposeSystem";
+        final String purposeSystemName2 = "purposeSystemName";
+        final String purposeDisplay2 = "purposeDisplay";
+        final Attribute attribute = OpenSAML2ComponentBuilder.getInstance().createPurposeOfUseAttribute(purposeCode2,
+            purposeSystem2, purposeSystemName2, purposeDisplay2);
+        try {
+            attribute.getAttributeValues().get(0).setDOM(getElementForSamlFile(getTestFilePath("complete_saml.xml")));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        helper.extractNhinCodedElement(attribute, CodeId);
+
+    }
+
+    /**
+     * Gets Saml File.
+     *
+     * @param filename the name of the file to retrieve
+     * @return
+     */
+
+    private File getSamlFile(String samlFileName) {
+        URI uri = null;
+        try {
+            uri = this.getClass().getResource(samlFileName).toURI();
+        } catch (URISyntaxException e) {
+            fail("Could not build URI for filepath. " + e.getMessage());
+        }
+        return new File(uri);
+    }
+
+    /**
+     * Gets the Element for Saml File.
+     *
+     * @param filename the name of the file to retrieve
+     * @return
+     */
+    private Element getElementForSamlFile(String samlFileName) throws Exception {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setNamespaceAware(true);
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        Document document = documentBuilder.parse(getSamlFile(samlFileName));
+        return document.getDocumentElement();
+    }
+
+    /**
+     * Gets the file path to the test resource.
+     *
+     * @param filename the name of the file to retrieve
+     * @return
+     */
+    private String getTestFilePath(String filename) {
+        // the first "/" is intentionally not using File.separator due to differences in how windows and unix based
```

Comment:
This is a strange comment on here.  Was this copied from another class?  Gateway JIRA tickets haven't been used in years.